test/dynamo/test_dynamo.py

```diff
@@ -103,6 +103,52 @@ def test_simple_model(self):
     self.assertNotIn('XLAData: None',
                      torch_xla._XLAC._get_xla_tensor_debug_info(xla_y3))
 
+  # Tests that the dynamo bridge automatically moves tensors to XLA device,
+  # then back to the original device.
+  @unittest.skipIf(xr.device_type() != "CUDA",
```

Comment:
why cuda only? In your implementation you didn;t have any guard on only run data moving logic only for cuda. If the intention is to do this auto-move for cuda, then it should throw an error or something if device is no cpu for example.

Reply:
See @vanbasten23 comments where his review said to update the test for CUDA. I think the intention should be to auto-move for any non-XLA device, and not limit it to CUDA. 

I can also replicate this test for a CPU version as well.

Reply:
Done with CUDA only.